modules/scope3/rtd/module.go

```diff
@@ -0,0 +1,456 @@
+	if segmentStore, ok := miCtx.ModuleContext["segments"].(*sync.Map); ok {
+		if val, ok := segmentStore.Load("segments"); ok {
+			segments = val.([]string)
+		}
+	}
+
+	if len(segments) == 0 {
+		return hookstage.HookResult[hookstage.AuctionResponsePayload]{}, nil
+	}
+
+	// Add segments to the auction response
+	changeSet := hookstage.ChangeSet[hookstage.AuctionResponsePayload]{}
+	changeSet.AddMutation(
+		func(payload hookstage.AuctionResponsePayload) (hookstage.AuctionResponsePayload, error) {
+			// Add Scope3 segments to the response ext so publisher can use them
+			if payload.BidResponse.Ext == nil {
+				payload.BidResponse.Ext = json.RawMessage("{}")
+			}
+
+			var extMap map[string]interface{}
+			if err := json.Unmarshal(payload.BidResponse.Ext, &extMap); err != nil {
+				extMap = make(map[string]interface{})
+			}
+
+			// Add segments as individual targeting keys for GAM integration
+			if m.cfg.AddToTargeting {
+				if prebidMap, ok := extMap["prebid"].(map[string]interface{}); ok {
+					if targetingMap, ok := prebidMap["targeting"].(map[string]interface{}); ok {
+						// Add each segment as individual targeting key
+						for _, segment := range segments {
+							targetingMap[segment] = "true"
+						}
+					} else {
+						// Create targeting map with individual segment keys
+						newTargeting := make(map[string]interface{})
+						for _, segment := range segments {
+							newTargeting[segment] = "true"
+						}
+						prebidMap["targeting"] = newTargeting
+					}
+				} else {
+					// Create prebid map with targeting
+					newTargeting := make(map[string]interface{})
+					for _, segment := range segments {
+						newTargeting[segment] = "true"
+					}
+					extMap["prebid"] = map[string]interface{}{
+						"targeting": newTargeting,
+					}
+				}
+			}
+
+			// Always add to a dedicated scope3 section for publisher flexibility
+			extMap["scope3"] = map[string]interface{}{
+				"segments": segments,
+			}
+
+			payload.BidResponse.Ext, _ = json.Marshal(extMap)
+			return payload, nil
+		},
+		hookstage.MutationUpdate,
+		"ext",
+	)
+
+	return hookstage.HookResult[hookstage.AuctionResponsePayload]{
+		ChangeSet: changeSet,
+	}, nil
+}
+
+// fetchScope3Segments calls the Scope3 API and extracts segments
+func (m *Module) fetchScope3Segments(ctx context.Context, bidRequest *openrtb2.BidRequest) ([]string, error) {
+	// Create cache key based on relevant user identifiers and site context
+	cacheKey := m.createCacheKey(bidRequest)
+
+	// Check cache first
+	if segments, found := m.cache.get(cacheKey, time.Duration(m.cfg.CacheTTL)*time.Second); found {
+		return segments, nil
+	}
+
+	// Enhance request with available user identifiers
+	m.enhanceRequestWithUserIDs(bidRequest)
+
+	// Marshal the bid request
+	requestBody, err := json.Marshal(bidRequest)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create HTTP request
+	req, err := http.NewRequestWithContext(ctx, "POST", m.cfg.Endpoint, bytes.NewReader(requestBody))
```

Comment:
if we're going to be make a lot of calls, we should use a `Transport` with better defaults